src/tests/test-helpers.hh

```diff
@@ -63,27 +63,31 @@ protected:
         int player_id_2 = 42;
         utils::Logger::get().level() = utils::Logger::DEBUG_LEVEL;
         auto players_ptr = make_players(player_id_1, player_id_2);
-        st = make_test_gamestate(some_map, players_ptr);
+        auto st = make_test_gamestate(some_map, players_ptr);
         players[0].id = player_id_1;
-        players[0].api = new Api(st, players_ptr->players[0]);
+        players[0].api = std::make_unique<Api>(
+            std::unique_ptr<GameState>(st->copy()), players_ptr->players[0]);
         players[1].id = player_id_2;
-        players[1].api = new Api(st, players_ptr->players[1]);
+        players[1].api = std::make_unique<Api>(
+            std::unique_ptr<GameState>(st->copy()), players_ptr->players[1]);
```

Comment:
I do not understand how can this work. How can the two players share the same GameState, if they have both a copy with their own ownership.

Reply:
Hum OK, it is not shared, but there is a clever mecanism of synchronisation.